src/puppetlabs/trapperkeeper/services/status/status_proxy_service.clj

```diff
@@ -8,15 +11,20 @@
    [:ConfigService get-in-config]]
   (init [this context]
     (log/info "Initializing status service proxy")
-    (let [target-url (URL. (get-in-config [:status-proxy :proxy-target-url]))
-          host (.getHost target-url)
-          port (.getPort target-url)
-          path (.getPath target-url)]
-      (add-proxy-route
-        this
-        {:host host
-         :port port
-         :path path}
-        {:ssl-config (get-in-config [:status-proxy :ssl-opts])
-         :scheme     :https}))
+    (let [status-proxy-config (get-in-config [:status-proxy])]
+      (schema/validate StatusProxyConfig status-proxy-config)
```

Comment:
I'm wondering if we should extract most of this work into a function in the core namespace? I know we've been trying to keep service function definitions as small as possible since they're really hard to debug because macros, and it seems like we're doing enough work here that it would be worth extracting.